Méthode reduce() - issue #13

melvinmajor opened the pull request.

melvinmajor committed:
Add reduce method

vvandenschrieck commented on src/main/java/tp/Fraction.java:
```diff
@@ -73,7 +73,18 @@ public void setDenominator(int denominator) {
+		if(getDenominator() == 1) {
+			gcm = 1;
+			reducedForm = String.valueOf(getNumerator());
+		}
```
Modifie le code de manière à modifier les variables d'instance numerator et denominator.  Attention aussi à couvrir tous les cas de figure au niveau du dénominateur.

melvinmajor committed:
Numerator and denominator are now modified

Numerator and denominator are modified based on the performed calculation.
More cases are taken into account.

melvinmajor commented:
Dans le cas du dénominateur à 0, une erreur est censé être reporté.
Faut-il l'implémenter sachant que Exception n'est pas implémenté et qu'aucune classe existante dans Java n'est extends à notre classe ?

vvandenschrieck commented on src/main/java/tp/Fraction.java:
```diff
@@ -74,16 +74,18 @@ public void setDenominator(int denominator) {
 		// Greatest common denominator
-		if(getDenominator() > 1) {
+		if(getDenominator() > 1 || getDenominator() < 0) {
 			gcm = getNumerator() % getDenominator();
```
Non, l'algorithme de calcul du PGCD est plus complexe que ça.

vvandenschrieck commented on src/main/java/tp/Fraction.java:
```diff
@@ -74,16 +74,18 @@ public void setDenominator(int denominator) {
-			reducedForm = String.valueOf(getNumerator());
+		else if(getDenominator() == 1) {
+			//Nothing to do, already the reduced form
+		}
```
Je ne comprend en fait pas vraiment la raison d'être de tes if.  Le dénominateur ne peut pas être 0, le code des constructeurs et des setters doit l'en empêcher.  Et s'il vaut 1, il devrait être géré dans le cas générique je pense.  De toute façon, un if sans instructions à l'intérieur n'est pas pertinent.

melvinmajor committed:
Simplified version
